### PR TITLE
Add post about SRE on sprint 80

### PR DESCRIPTION
**Pull request by saraycp**
Co-authored-by: David Kang <dkang@suse.com>

### CODE REVIEW

**Comment by hennevogel** [comment]
Deploy preview for *openbuildservice* ready!

Built with commit f5ec744b80e8dc579c777be33dae3394cbe7e372

https://deploy-preview-266--openbuildservice.netlify.app

File: _posts/development/2020-07-31-we-do-sre.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+layout: post
+title: Yes, we do SRE!
+category: development
+---
+
+How do you think we keep our reference instance up and running?
+
+Of course, we do Site Reliability Engineering (SRE).
```
Comment (hennevogel) [inline _posts/development/2020-07-31-we-do-sre.md:9]: I don't think many people who read this blog (our users) do operations. You would have to explain what this "name" means.

File: _posts/development/2020-07-31-we-do-sre.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+We always have an eye on what's going on with the hosts, services and applications we are responsible for.
+And you, people from the community, can also do so. OBS is public and so are some of the dashboards we use to ensure Site Reliability.
```
Comment (hennevogel) [inline _posts/development/2020-07-31-we-do-sre.md:11]: That sounds as if *we* would not be "from the community". We are part of the community.

File: _posts/development/2020-07-31-we-do-sre.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+We take the data we need to do SRE from five different points:
+
+* Logging
+* Exception Tracking
+* System/Services Monitoring
+* Application Performance Monitoring (APM)
+* Application Health Monitoring (AHM)
+
+You can read more about all of them starting [here](https://github.com/openSUSE/open-build-service/wiki/Site-Reliability).
```
Comment (hennevogel) [inline _posts/development/2020-07-31-we-do-sre.md:21]:
```suggestion
You can read more about all of them starting [in our github wiki](https://github.com/openSUSE/open-build-service/wiki/Site-Reliability).
```

File: _posts/development/2020-07-31-we-do-sre.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+During the last few weeks we have deeply analyzed what we are measuring
+and we have improved our documentation regarding this topic.
```
Comment (hennevogel) [inline _posts/development/2020-07-31-we-do-sre.md:24]: Why have we done this?

File: _posts/development/2020-07-31-we-do-sre.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+We have also set up some new alerts from the different dashboards.
+From now on we are better informed when something goes wrong in our reference instance.
+
+<img src="/images/posts/sprint_80_grafana_panel.png" alt="Screenshot of Grafana panel"/>
+
+Stay tuned, we will share with you any achievements on this area.
```
Comment (krauselukas) [inline _posts/development/2020-07-31-we-do-sre.md:30]: @saraycp dont forget to include the little partial about the feedback `{% include partials/_how-to-give-us-feedback.md %}`

File: _posts/development/2020-07-31-we-do-sre.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+---
+layout: post
+title: Yes, we do SRE!
+category: development
+---
+
+How do you think we keep our reference instance up and running?
```
Comment (dmarcoux) [inline _posts/development/2020-07-31-we-do-sre.md:7]: We're missing the [lede](https://en.wikipedia.org/wiki/Five_Ws).

File: _posts/development/2020-07-31-we-do-sre.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+* Logging
+* Exception Tracking
+* System/Services Monitoring
+* Application Performance Monitoring (APM)
+* Application Health Monitoring (AHM)
+
+You can read more about all of them starting [here](https://github.com/openSUSE/open-build-service/wiki/Site-Reliability).
+
+During the last few weeks we have deeply analyzed what we are measuring
```
Comment (dmarcoux) [inline _posts/development/2020-07-31-we-do-sre.md:23]:
```suggestion
During the last few weeks, we have deeply analyzed what we are measuring
```

File: _posts/development/2020-07-31-we-do-sre.md (hunk 1 above)
Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md]: Corrected

File: _posts/development/2020-07-31-we-do-sre.md (hunk 2 above)
Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md]: Rephrased.

File: _posts/development/2020-07-31-we-do-sre.md (hunk 3 above)
Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md]: Done

File: _posts/development/2020-07-31-we-do-sre.md (hunk 4 above)
Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md]: Some reasons have been added at the beginning of the section.

File: _posts/development/2020-07-31-we-do-sre.md (hunk 5 above)
Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md]: Done

File: _posts/development/2020-07-31-we-do-sre.md (hunk 6 above)
Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md:7]: I revisit the whole document, do you still think _lede_ is not obvious in the text?

File: _posts/development/2020-07-31-we-do-sre.md (hunk 7 above)
Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md]: Done

File: _posts/development/2020-07-31-we-do-sre.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+category: development
+---
+
+How do you think we keep our reference instance up and running?
+
+We practice Site Reliability Engineering (SRE).
+
+## Operations & SRE
+
+Deployment, continuous integration or monitoring are well-known terms we inmediatelly connect with operations.
```
Comment (dmarcoux) [inline _posts/development/2020-07-31-we-do-sre.md:13]:
```suggestion
Deployment, continuous integration or monitoring are well-known terms we immediately connect with operations.
```

File: _posts/development/2020-07-31-we-do-sre.md (hunk 6 above)
Comment (dmarcoux) [reply _posts/development/2020-07-31-we-do-sre.md:7]: Before going into details about SRE in the various sections below, we want to communicate:

- Who: we (the Build Solutions team)
- What: practice Site Reliability Engineering (SRE)
- When: ???
- Where: ???
- Why: to keep our reference instance up and running

Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md:7]: Thanks for your explanation @dmarcoux , I think I understand now. What do you think about this?
https://github.com/openSUSE/obs-landing/pull/266/files#diff-32d86a6a39e1811f72ad3f4ec570c8f5R7-R13

Comment (dmarcoux) [reply _posts/development/2020-07-31-we-do-sre.md:7]: Good work :)

Sorry for not directly pointing out to what I meant in my first comment.

File: _posts/development/2020-07-31-we-do-sre.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+We can say SRE tries to manage IT operations efficiently.
+Where DevOps focuses on ensuring an effective deployment and avoiding system downtime, among others,
+SRE focuses on the post-failure situation analyzing the causes and aiming to maximize uptime and stability.
+
+After deployment, SRE makes use of monitoring tools to find out where automation is crucial to improving a system’s health and availability.
+
+## SRE in OBS
+
+Introducing SRE into the OBS operations makes it possible for us to know and manage the state of the deployment,
```
Comment (krauselukas) [inline _posts/development/2020-07-31-we-do-sre.md:28]:
```suggestion
Introducing SRE into the OBS operations makes it possible for us to know and manage the state of the deployments. Through continuous deployment's we have

```

File: _posts/development/2020-07-31-we-do-sre.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+We can say SRE tries to manage IT operations efficiently.
+Where DevOps focuses on ensuring an effective deployment and avoiding system downtime, among others,
+SRE focuses on the post-failure situation analyzing the causes and aiming to maximize uptime and stability.
+
+After deployment, SRE makes use of monitoring tools to find out where automation is crucial to improving a system’s health and availability.
+
+## SRE in OBS
+
+Introducing SRE into the OBS operations makes it possible for us to know and manage the state of the deployments,
```
Comment (krauselukas) [inline _posts/development/2020-07-31-we-do-sre.md:28]:
```suggestion
Introducing SRE into the OBS operations makes it possible for us to know and manage the state of the deployments.
```

File: _posts/development/2020-07-31-we-do-sre.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+through continuous deployment have quicker feedback loops about the code we have deployed and hence avoid deployment downtime in the future.
```
Comment (krauselukas) [inline _posts/development/2020-07-31-we-do-sre.md:29]:
```suggestion
Through continuous deployment's we've quicker feedback loops about the code we have deployed and hence avoid  downtimes in the future.
```

Comment (krauselukas) [reply _posts/development/2020-07-31-we-do-sre.md:29]: To many times the word 'deploy', maybe the last one is not needed :)

Comment (dmarcoux) [reply _posts/development/2020-07-31-we-do-sre.md]: If the line 28 is its own sentence, I would suggest this instead:

```suggestion
Through continuous deployment, we have quicker feedback loops about the code we have deployed and hence avoid deployment downtime in the future.
```

Abbreviating the verb `have` when it's not an auxiliary is quite uncommon. So I would be against the suggestion from @krauselukas.

`deployment's` is wrong. `'s` is added for possessive nouns and it's not the case here. Have a look at this [article](https://www.thepunctuationguide.com/apostrophe.html) on the subject.

File: _posts/development/2020-07-31-we-do-sre.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+SRE focuses on the post-failure situation analyzing the causes and aiming to maximize uptime and stability.
+
+After deployment, SRE makes use of monitoring tools to find out where automation is crucial to improving a system’s health and availability.
+
+## SRE in OBS
+
+Introducing SRE into the OBS operations makes it possible for us to know and manage the state of the deployments.
+Through continuous deployments we have quicker feedback loops about the code we have deployed and hence avoid deployment downtime in the future.
+
+It also puts the OBS on the path of observability where the team is able to ask arbitrary questions about our app/deployment without - and this is the key part - having to know ahead of time what we wanted to ask. The deeper we drill into the OBS code base, the more complex the issues will become, the more we will encounter unknown unknowns. Problems we never have faced before that require the team to ask questions they have never asked before. And the infrastructure needs to support this.
```
Comment (krauselukas) [inline _posts/development/2020-07-31-we-do-sre.md:31]:
```suggestion
It also puts the OBS on the path of observability where the team is able to ask arbitrary questions about our app/deployment without - and this is the key part - having to know ahead of time what we need to ask for. The deeper we drill into the OBS code base, the more complex the issues will become, the more we will encounter unknown unknowns. Problems we never have faced before that require the team to ask questions they have never asked before. And the infrastructure needs to support this.
```

Comment (krauselukas) [reply _posts/development/2020-07-31-we-do-sre.md:31]: @saraycp last suggestion :sweat_smile:

File: _posts/development/2020-07-31-we-do-sre.md (hunk 11 above)
Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md]: Exactly. I corrected as Dany suggested.
@krauselukas I dropped the last "deployment" word, I think is ok anyway.

File: _posts/development/2020-07-31-we-do-sre.md (hunk 12 above)
Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md:31]: I wouldn't change it.

Comment (saraycp) [reply _posts/development/2020-07-31-we-do-sre.md]: Changed

File: _posts/development/2020-07-31-we-do-sre.md (hunk 11 above)
Comment (krauselukas) [reply _posts/development/2020-07-31-we-do-sre.md]: :+1:

**Pull request merged**